tests/snap_test.go

```diff
@@ -0,0 +1,201 @@
+	if err != nil || stderr != "" {
+		log.Printf("stderr: %s", stderr)
+		log.Printf("err: %s", err)
+		return "", err
+	}
+
+	utils.Exec(nil, "rm -rf squashfs-root")
+	return newPath, nil
+}
+
+func setup() (teardown func(), err error) {
+	var newPath string
+
+	log.Println("[CLEAN]")
+	utils.SnapRemove(nil, snapName)
+
+	log.Println("[SETUP]")
+
+	teardown = func() {
+		log.Println("[TEARDOWN]")
+		utils.SnapDumpLogs(nil, start, snapName)
+
+		utils.Exec(nil, "rm "+newPath)
+
+		log.Println("Removing installed snap:", !utils.SkipTeardownRemoval)
+		if !utils.SkipTeardownRemoval {
+			utils.SnapRemove(nil, snapName)
+		}
+	}
+
+	// authorize gpio mock
+	if utils.LocalServiceSnap() {
+		newPath, err = authorizeGpioMock(utils.LocalServiceSnapPath)
+	}
+	if err != nil {
+		teardown()
+		return
+	}
+
+	// install matter-pi-gpio-commander
+	if utils.LocalServiceSnap() {
+		err = utils.SnapInstallFromFile(nil, newPath)
+	} else {
+		err = utils.SnapInstallFromStore(nil, snapName, utils.ServiceChannel)
+	}
+	if err != nil {
+		teardown()
+		return
+	}
+
+	if err = setupGPIO(); err != nil {
+		teardown()
+		return
+	}
+
+	// connect interfaces:
+	utils.SnapConnect(nil, snapName+":avahi-control", "")
+	utils.SnapConnect(nil, snapName+":bluez", "")
+	utils.SnapConnect(nil, snapName+":network", "")
+	utils.SnapConnect(nil, snapName+":network-bind", "")
+	utils.SnapConnect(nil, snapName+":custom-gpio", snapName+":custom-gpio-dev")
+
+	return
+}
+
+func getMockGPIO() (string, error) {
+	gpioChipNumber, stderr, err := utils.Exec(nil, "ls /dev/gpiochip* | sort -n | tail -n 1 | cut -d'p' -f3")
+	if err != nil || stderr != "" {
+		log.Printf("Failed to get mock gpio chip number, Error %s: %s", stderr, err)
+		return "", err
+	}
+	return gpioChipNumber, nil
+}
+
+func setupGPIO() error {
+
+	TestedGpioChip := os.Getenv(specificGpioChip)
+	TestedGpioLine := os.Getenv(specifGpioLine)
+
+	gpiochip := TestedGpioChip
+	var err error
+	if gpiochip == "" {
+		gpiochip, err = getMockGPIO()
+		if err != nil {
+			log.Printf("Failed to get mock gpio chip number: %s", err)
+			return err
+		}
+
+		log.Printf("[TEST] No specific gpiochip defined, using mockup gpio: /dev/gpiochip%s", gpiochip)
+	}
+
+	gpioline := TestedGpioLine
+	if gpioline == "" {
+		gpioline = "4"
+		log.Printf("[TEST] No specific gpio line defined, using default: %s", gpioline)
+	}
+
+	utils.SnapSet(nil, snapName, "gpiochip", gpiochip)
+	utils.SnapSet(nil, snapName, "gpio", gpioline)
+
+	return nil
+}
+
+func TestBlinkOperation(t *testing.T) {
+	log.Println("[TEST] Standard blink operation")
+
+	// test blink operation
+	t.Run("test-blink", func(t *testing.T) {
+		ctx, cancel := context.WithDeadline(context.Background(), <-time.After(20*time.Second))
+		defer cancel()
+
+		stdout, _, err := utils.ExecContextVerbose(nil, ctx, snapName+".test-blink")
+		t.Logf("err: %s", err)
+		t.Logf("stdout: %s", stdout)
+		assert.Error(t, err, "Expected an error")
```

Comment:
`assert.Error` logs will show what was expected. An extra message isn't needed unless if it adds more info.